packages/hardhat/scripts/new-dao.ts

```diff
@@ -0,0 +1,220 @@
+import hre, { ethers } from "hardhat";
+import { Contract } from "@ethersproject/contracts";
+import { Signer } from "@ethersproject/abstract-signer";
+import { GardensTemplate, Kernel } from "../typechain";
+
+const { deployments } = hre;
+
+const network = hre.network.name !== 'localhost' ? hre.network.name :  'xdai'
+const blockTime = network === "rinkeby" ? 15 : network === "mainnet" ? 13 : 5; // 15 rinkeby, 13 mainnet, 5 xdai
+
+console.log(`Every ${blockTime}s a new block is mined in ${network}.`);
+
+const ONE_HUNDRED_PERCENT = 1e18
+const ISSUANCE_ONE_HUNDRED_PERCENT = 1e10
+const CONVICTION_VOTING_ONE_HUNDRED_PERCENT = 1e7
+const ONE_TOKEN = 1e18
+const ONE_MINUTE = 60
+const ONE_HOUR = 60 * ONE_MINUTE
+const ONE_DAY = 24 * ONE_HOUR
+const ONE_YEAR = 365 * ONE_DAY
+
+const getApps = async (daoAddress: string, appIds: string[]): Promise<string[]> => {
+  const dao = (await ethers.getContractAt("Kernel", daoAddress)) as Kernel;
+  const apps = await dao.queryFilter(dao.filters.NewAppProxy(null, null, null))
+    .then(events => events
+      .filter(({ args }) => appIds.includes(args.appId))
+      .map(({ args }) => ({
+        appId: args.appId,
+        proxy: args.proxy
+      }))
+      .reduce((apps, { appId, proxy }) => ({ ...apps, [appId]: !apps[appId] ? proxy : [...apps[appId], proxy] }), {})
+    )
+  return appIds.map(appId => apps[appId])
+}
+
+const gardensTemplateAddress = async (): Promise<string> => (await deployments.get("GardensTemplate")).address;
+
+const getGardensTemplate = async (signer: Signer): Promise<GardensTemplate> =>
+  (await ethers.getContractAt("GardensTemplate", await gardensTemplateAddress(), signer)) as GardensTemplate;
+
+const getAddress = async (selectedFilter: string, contract: Contract, transactionHash: string): Promise<string> => {
```

Comment:
This isn't an issue but I'm curious what the best way of retrieving an arbitrary argument from an event is?

Reply:
Done :+1: